File: Weekly Contest 352/2761. Prime Pairs With Target Sum.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<List<Integer>> findPrimePairs(int n) {
        // First calculate the prime numbers
        boolean[] prime = new boolean[n + 1];
        for (int i = 2; i <= n; i++) {
            prime[i] = true;
        }

        for (int i = 2; i <= n; i++) {
            if (!prime[i]) {
                continue;
            }
            for (int j = 2 * i; j <= n; j += i) {
                prime[j] = false;
            }
        }

        // Iterate and find the pairs whose sum is equal to n
        List<List<Integer>> ans = new ArrayList<>();
        for (int i = 2; 2 * i <= n; i++) {
            if (prime[i] && prime[n - i]) {
                List<Integer> pair = new ArrayList<>();
                pair.add(i);
                pair.add(n - i);
                ans.add(pair);
            }
        }

        return ans;
    }
}

File: Weekly Contest 352/2761. Prime Pairs With Target Sum.py
class Solution:
    def findPrimePairs(self, n: int) -> List[List[int]]:
        # First calculate the prime numbers
        prime = [True] * (n + 1)
        prime[0] = prime[1] = False

        for i in range(2, n + 1):
            if prime[i]:
                for j in range(2 * i, n + 1, i):
                    prime[j] = False

        # Iterate and find the pairs whose sum is equal to n
        ans = []
        for i in range(2, n // 2 + 1):
            if prime[i] and prime[n - i]:
                ans.append([i, n - i])

        return ans
